static/extensions/MubiLop/penguinhook.js

```diff
@@ -0,0 +1,258 @@
+                            }
+                        }
+                    },
+                    {
+                        opcode: 'sendWebhookWithEmbed',
+                        text: 'send webhook with embeds to webhook [WEBHOOK]',
+                        blockType: Scratch.BlockType.COMMAND,
+                        arguments: {
+                            WEBHOOK: {
+                                type: Scratch.ArgumentType.STRING,
+                                defaultValue: 'https://your.webhook.url'
+                            }
+                        }
+                    },
+                    {
+                        blockType: Scratch.BlockType.LABEL,
+                        text: 'Customizable Section'
+                    },
+                    {
+                        opcode: 'sendjson',
+                        blockType: Scratch.BlockType.COMMAND,
+                        text: 'send using json [JSON_DATA] webhook url: [WEBHOOK]',
+                        arguments: {
+                            JSON_DATA: {
+                                type: Scratch.ArgumentType.STRING,
+                                defaultValue: '{"content": "Sent with JSON!!","embeds": null,"attachments": []}'
+                            },
+                            WEBHOOK: {
+                                type: Scratch.ArgumentType.STRING,
+                                defaultValue: 'https://your.webhook.url'
+                            }
+                        }
+                    }
+                ]
+            };
+        }
+
+        // New functions for setting individual embed properties
+        setTitle(args) {
+            this.embedProperties.title = Scratch.Cast.toString(args.TITLE);
+        }
+
+        setDescription(args) {
+            this.embedProperties.description = Scratch.Cast.toString(args.DESCRIPTION);
+        }
+
+        setFooter(args) {
+            this.embedProperties.footer = Scratch.Cast.toString(args.FOOTER);
+        }
+
+        setHexColor(args) {
+            this.embedProperties.hexColor = Scratch.Cast.toString(args.COLOR);
+        }
+
+        setName(args) {
+            this.embedProperties.name = Scratch.Cast.toString(args.NAME);
+        }
+
+        sendWebhook(args) {
+            const { NAME, IMAGEURL, MESSAGE, WEBHOOK } = args;
+            const name = Scratch.Cast.toString(NAME);
+            const imageUrl = Scratch.Cast.toString(IMAGEURL);
+            const message = Scratch.Cast.toString(MESSAGE);
+            const webhookUrl = Scratch.Cast.toString(WEBHOOK);
+            let nameReq;
+
+            if (name) {
+                nameReq = name;
+            }
+
+            const payload = {
+                username: nameReq,
+                avatar_url: imageUrl,
+                content: message
+            };
+
+            console.log(JSON.stringify(payload))
+
+            return fetch(webhookUrl, {
```

Comment:
Scratch.canFetch() is not checked before this, meaning this will just send data to a webhook without the user knowing at all